codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/UnionGenerator.java

```diff
@@ -51,48 +54,56 @@ public void run() {
             writer.write("is$L()", symbol.getName());
         }).write("");
 
-        // Create type aliases for each member that satisfy the interface.
+        // Create structs for each member that satisfy the interface.
         for (MemberShape member : shape.getAllMembers().values()) {
             Symbol memberSymbol = symbolProvider.toSymbol(member);
-            String exportedMemberName = symbol.getName() + symbolProvider.toMemberName(member);
-            String unExportedMemberName = StringUtils.uncapitalize(exportedMemberName);
+            String exportedMemberName = String.format(
+                    "%sMember%s", symbol.getName(), symbolProvider.toMemberName(member));
```

Comment:
would be nice to be a helper function for this pattern, so easier to ensure its consistent.